### ISSUE TITLE
Line length option ignored on startup

### ISSUE THREAD

**Issue by Jakob-Bach** [spyder-ide/spyder#13734]
## Problem Description

Under "Preferences" > "Completion and linting" > "Code style", I have ticked "Enable code style linting". Also, I have changed the option "Maximum allowed line length" by increasing the allowed line length. Whenever I start Spyder, this line length option is ignored and I get lots of warnings in the editor of violating the (default) line length of 79 characters. I need to open the "Code style" settings, edit some option (in any way, even changing it and then immediately setting it back is sufficient) and then press "OK" or "Apply" so the line length option is picked up.

### What steps reproduce the problem?

1. Enable linting (with default settings)
2. Create a file which has lines longer than 79 characters; code style warnings appear
3. Change "Maximum allowed line length" to accommodate for the new line length; code style warnings disappear
4. Restart Spyder; code style warnings appear
5. Change an arbitrary code style setting, change it back and press "OK" or "Apply"; code style warnings disappear

### What is the expected output? What do you see instead?

The line length option should be picked up from the start.

## Versions

* Spyder version:  4.1.5
* Python version: 3.7.7 64-bit
* Qt version: 5.9.6
* PyQt version: 5.9.2
* Operating System name/version: Windows 10

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0           :  1.4.0 (OK)
chardet >=2.0.0                :  3.0.4 (OK)
cloudpickle >=0.5.0            :  1.6.0 (OK)
diff_match_patch >=20181111    :  20200713 (OK)
intervaltree                   :  None (OK)
IPython >=4.0                  :  7.18.1 (OK)
jedi =0.17.1                   :  0.17.1 (OK)
nbconvert >=4.0                :  5.6.1 (OK)
numpydoc >=0.6.0               :  1.1.0 (OK)
paramiko >=2.4.0               :  2.7.2 (OK)
parso =0.7.0                   :  0.7.0 (OK)
pexpect >=4.4.0                :  4.8.0 (OK)
pickleshare >=0.4              :  0.7.5 (OK)
psutil >=5.3                   :  5.7.2 (OK)
pygments >=2.0                 :  2.6.1 (OK)
pylint >=1.0                   :  2.4.4 (OK)
pyls >=0.34.0;<1.0.0           :  0.34.1 (OK)
qdarkstyle >=2.8               :  2.8.1 (OK)
qtawesome >=0.5.7              :  0.7.2 (OK)
qtconsole >=4.6.0              :  4.7.6 (OK)
qtpy >=1.5.0                   :  1.9.0 (OK)
rtree >=0.8.3                  :  0.9.4 (OK)
sphinx >=0.6.6                 :  3.2.1 (OK)
spyder_kernels >=1.9.4;<1.10.0 :  1.9.4 (OK)
watchdog                       :  None (OK)
zmq >=17                       :  19.0.1 (OK)

# Optional:
cython >=0.21                  :  0.29.21 (OK)
matplotlib >=2.0.0             :  3.3.1 (OK)
numpy >=1.7                    :  1.19.1 (OK)
pandas >=0.13.1                :  1.1.1 (OK)
scipy >=0.17.0                 :  1.5.2 (OK)
sympy >=0.7.3                  :  1.6.2 (OK)
```

**Comment by steff456** [comment]
Hi @Jakob-Bach,

I was unable to reproduce this issue with the steps you wrote. Can you please upload a gif with the behavior you are having?

**Comment by Jakob-Bach** [comment]
Hi @steff456, I created a gif. I also noticed that this problem only occurs if Spyder automatically opens a project on startup (though this does not apply to one specific project, but all my projects). If I finished the last Spyder session without an open project and then start Spyder, the problem does not occur (even if I then directly open a project after startup).

![Spyder](https://user-images.githubusercontent.com/6431179/92912672-4b753e00-f42a-11ea-8e07-9cf9581f9e3f.gif)
